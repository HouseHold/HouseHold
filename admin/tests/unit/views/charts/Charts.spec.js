import Vue from 'vue'
import { shallowMount } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue'
import Charts from '@/views/charts/Charts'

Vue.use(CoreuiVue)

describe('Charts.vue', () => {
  it('has a name', () => {
    expect(Charts.name).toBe('Charts')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Charts)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Charts', () => {
    const wrapper = shallowMount(Charts)
    expect(wrapper.is(Charts)).toBe(true)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Charts)
    expect(wrapper.element).toMatchSnapshot()
  })
})
