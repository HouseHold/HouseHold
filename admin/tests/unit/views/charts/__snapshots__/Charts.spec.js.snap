// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts.vue renders correctly 1`] = `
<div>
  <ccardgroup-stub
    class="card-columns cols-2"
    columns="true"
    tag="div"
  >
    <ccard-stub>
      <ccardheader-stub>
        
        Line Chart
        
        <div
          class="card-header-actions"
        >
          <a
            class="card-header-action"
            href="https://coreui.io/vue/docs/components/charts"
            rel="noreferrer noopener"
            target="_blank"
          >
            <small
              class="text-muted"
            >
              docs
            </small>
          </a>
        </div>
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartlineexample-stub />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Bar Chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartbarexample-stub />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Doughnut Chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartdoughnutexample-stub />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Radar Chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartradarexample-stub />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Pie Chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartpieexample-stub />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Polar Area Chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartpolarareaexample-stub />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Simple line chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartlinesimple-stub
          backgroundcolor="transparent"
          bordercolor="success"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          labels="months"
        />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Simple pointed chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartlinesimple-stub
          backgroundcolor="transparent"
          bordercolor="warning"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
          pointed="true"
        />
      </ccardbody-stub>
    </ccard-stub>
     
    <ccard-stub>
      <ccardheader-stub>
        Simple bar chart
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cchartbarsimple-stub
          backgroundcolor="danger"
          datapoints="10,22,34,46,58,70,46,23,45,78,34,12"
          label="Sales"
        />
      </ccardbody-stub>
    </ccard-stub>
  </ccardgroup-stub>
</div>
`;
